native/core/src/execution/datafusion/planner.rs

```diff
@@ -1775,10 +1780,14 @@ impl From<ExpressionError> for DataFusionError {
 /// modification. This is used to determine if we need to copy the input batch to avoid
 /// data corruption from reusing the input batch.
 fn can_reuse_input_batch(op: &Arc<dyn ExecutionPlan>) -> bool {
-    op.as_any().is::<ScanExec>()
```

Comment:
Hmm, but our native scan actually reuses batches.

Reply:
yes, it is still being checked here:

```rust
 } else {
         op.as_any().is::<ScanExec>()
     }
```

Reply:
Ah, okay, I missed it.